daemon/daemon.go

```diff
@@ -1515,6 +1515,34 @@ func CreateDaemonRoot(config *config.Config) error {
 	return setupDaemonRoot(config, realRoot, idMapping.RootPair())
 }
 
+// RemapContainerdNamespaces returns the right containerd namespaces to use:
+// - if they are not already set in the config file
+// -  and the daemon is running with user namespace remapping enabled
+// Then it will return new namespace names, otherwise it will return the existing
+// namespaces
+func RemapContainerdNamespaces(config *config.Config) (string, string, error) {
+	idMapping, err := setupRemappedRoot(config)
```

Comment:
Curious are we potentially running this multiple times now?

<img width="514" alt="Screenshot 2024-01-24 at 14 08 56" src="https://github.com/moby/moby/assets/1804568/82ea77d2-1028-440c-be31-d9a959d8630c">


It looks like the new code may be called before the existing code; do we need the daemon root to be created at this point already, or do we only need the containerd namespaces to use? (and could we defer creating the root to the existing code? 🤔

Reply:
The function is poorly named, it doesn't create the root, it creates the idtools mapping struct

Reply:
Exactly what I said. https://github.com/moby/moby/pull/46786/files#r1464925717

Reply:
Moved

Reply:
Oh, LOL, missed that comment 😂